src/cdk/testing/component-harness.ts

```diff
@@ -421,3 +480,15 @@ export class HarnessPredicate<T extends ComponentHarness> {
     }
   }
 }
+
+/** Represent a value as a string for the purpose of logging. */
+function _valueAsString(value: unknown) {
+  if (value === undefined) {
+    return 'undefined';
+  }
+  // `JSON.stringify` doesn't handle RegExp properly, so we need a custom replacer.
+  return JSON.stringify(value, (_, v) =>
```

Comment:
`JSON.stringify` will fail if the structure has a cycle. Would it make sense to handle that here?

Reply:
Yeah I can just give up and say "{...}" if this happens